src/adapter/adapter.ts

```diff
@@ -0,0 +1,55 @@
+import couchbase from 'couchbase';
+import { extractConnectionString } from '../utils/extract-connection-string';
+
+export interface ConnectOptions {
+  connectionString: string;
+  username: string;
+  password: string;
+  bucketName: string;
+}
+
+class ConnectionManager {
```

Comment:
Note that the SDK itself handles management of cacheing of buckets/collections.

Reply:
Thank you @brett19 for your advice :+1: 

We plan to use all the functionalities already implemented in SDK 3

*This tips will be added in the next pull-request*